README.md

```diff
@@ -0,0 +1,126 @@
+# SolidusAssetsProxy
+
+This [Solidus](https://github.com/solidusjs/solidus) library provides an easy solution for proxying a site's assets with a CDN or a an image processing service like [imgix](https://www.imgix.com/) and [ReSRC](https://www.resrc.it/).
+
+This process can be complex since a site will typically use assets from many sources: local server, Amazon S3, URLs from the site's resources, etc. Making sure all URLs in the site's views use the proxy, and setting up the proxy for all these changing sources is time consuming and prone to error.
+
+SolidusAssetsProxy provides two solutions to this:
+ - Utility functions to change all assets URLs to proxy URLs in a Solidus context, to be used in a preprocessor. For example, all URLs matching `http://source.com/images/...` would be replaced by `http://proxy.com/mysite/[encoded asset URL]`. No need to setup multiple proxies, a single proxy will take care of all assets sources.
+ - A redirects server, that the proxy can use to retrieve the assets data. This server will decode the `[encoded asset URL]` part of the URL above, validate that it matches one of the configurable whitelisted hosts, and redirect to the asset URL. A single redirects server can be used by multiple Solidus sites, and even when developing a new site.
+
+## Installation
+
+```
+$ npm install solidus-assets-proxy --save
+```
+
+## Library Usage
+
+### Initialization
+
+```javascript
+var SolidusAssetsProxy = require('solidus-assets-proxy');
+var assets_proxy = new SolidusAssetsProxy([
+  {
+    origin: 'http://resource1.com/images',
+    proxy: 'http://mysite.proxy1.com'
+  },
+  {
+    origin: /http:\/\/resource(?:2|3)\.com\/pictures/, // Note the non-capturing group
+    proxy: 'http://proxy2.com/mysite'
```

Comment:
This seems a bit at odds with the statement "A single redirects server can be used by multiple Solidus sites, and even when developing a new site" that comes right before. Why not:

```js
new SolidusAssetsProxy({
  origins: ['http://resource1.com/images'],
  proxy: 'http://mysite.proxy1.com'
});
```

Deferring varying proxies until a valid use case for that materializes.